### PR TITLE
Update ResultProvider with breaking LMR changes

### PR DESCRIPTION
**Pull request by amcasey**
Microsoft.VisualStudio.Debugger.Metadata is going to be making some breaking changes.  These are the corresponding ResultProvider changes.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @cston @KevinH-MS @ManishJayaswal @pnelsonmsft

**Comment by amcasey** [comment]
As I mentioned offline, I expect the Jenkins run to fail because it won't have the new M.V.D.Metadata in the GAC.

**Comment by amcasey** [comment]
FYI @gregg-miskelly

File: src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs [hunk 1]
```diff
@@ -234,7 +235,11 @@ public void EvaluateDebuggerDisplayString(DkmWorkList workList, DkmInspectionCon
                     }
 
                     var type = ((TypeImpl)this.Type.GetLmrType()).Type;
-                    var bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static;
+                    var bindingFlags = 
+                        System.Reflection.BindingFlags.Public | 
+                        System.Reflection.BindingFlags.NonPublic | 
+                        System.Reflection.BindingFlags.Instance | 
+                        System.Reflection.BindingFlags.Static;
```
Comment (cston) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs]: const?

**Comment by cston** [comment]
LGTM

**Comment by KevinH-MS** [comment]
LGTM :shipit:

File: src/ExpressionEvaluator/Core/Source/Concord/Concord.csproj [hunk 2]
```diff
@@ -31,8 +31,12 @@
     <None Include="packages.config" />
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/Core/Source/Concord/Concord.csproj]: Consider adding `<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file after the xml declation.

File: src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs (hunk 1 above)
Comment (amcasey) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs]: Yep.

File: src/ExpressionEvaluator/Core/Source/Concord/Concord.csproj (hunk 2 above)
Comment (amcasey) [inline src/ExpressionEvaluator/Core/Source/Concord/Concord.csproj]: Sure.

**Pull request merged**